#2539: change condition from truthy to not nully

grovesNL opened the pull request:
As described in #2539, certain falsy values such as `0` should be acceptable. The truthy condition has been changed to check whether the value is `null` or `undefined` instead.

taion commented:
huh, weird build failure.

code looks fine to me. wonder if we should have a test – the concern here would be regressions, not bugs in this code specificall

grovesNL commented:
@taion I changed another condition that I had missed initially, and added two tests for acceptable number/string event keys. Let me know if it's sufficient.
